python/tank/bootstrap/manager.py

```diff
@@ -202,6 +203,28 @@ def _set_bundle_cache_fallback_paths(self, paths):
     )
 
 
+    def _get_async_bootstrap(self):
+        """
+        Boolean property that specifies how the engine bootstrap logic should be executed.
+
+        By default, when ```async_bootstrap`` is false, the whole engine bootstrap logic
+        will be executed synchronously in the main application thread.
+
+        When ``async_bootstrap` is true, an :class:`sgtk.Sgtk` instance will be bootstrapped asynchronously
+        in a background thread, followed by launching the engine synchronously
+        in the main application thread. This will allow the main application to continue
```

Comment:
I would not dare starting the Maya engine as currently coded in a background thread.

I like the idea of a per-engine property for async startup or not. This could neatly shield the plug-in client from all threading startup decisions. Though, a way to retrieve this property when only the engine name is known would have to be devised.